### PR TITLE
Port /employees (Remind Presidents) to warondisease.org

### PR DESCRIPTION
**Pull request by mikepsinn**
Closes the remaining piece of #243 (parent #238): the president management system now lives on the campaign domain, where the people it asks visitors to pressure already are.

## What this does

`/employees` on warondisease.org is **the composer plus the roster**, not Optimitron's task table.

Optimitron builds this page out of `getTasksPageData`, which loads the whole task tree so a generic table can rank, filter, fund, claim, and pay out every task in the system. A campaign visitor can reach none of that. This reads the one thing the page actually shows through `getTreatySignerTasks` — a single index range scan over the signer task-key prefix.

**Deliberately dropped with the table:**

- **The per-row remind dialog.** The composer at the top is the one call to action. A second remind button on every row splits it.
- **The live death and dollar counters.** They recompute on a timer, so every screenshot of the page differs from the last — the exact instability the visual-review harness keeps flagging on other PRs. Bringing them here would import that problem into a second app.

That is option A of the three I offered on #243. Option B kept the counters, option C ported `task-card`/`task-row`/`rank-tasks` whole — 31,816 lines of import closure versus 3,750 here. Say the word if you want either of the others and I will do it as a follow-up.

## Ordering — the one place the slim version had to earn its keep

All 189 signer tasks carry the same title (`Sign the 1% Treaty`) and the same due date. The database cannot order them meaningfully, and with no tiebreak the query planner is free to reshuffle the roster between renders — page 1 would show a different set of leaders on a refresh.

Sorting by the assignee government's **annual military spending** fixes both problems at once: it is the figure the 1% Treaty asks each government to redirect, so the biggest spenders are exactly the ones a visitor should be reminding, and it is deterministic. Assignee name then id breaks the remaining ties, so governments with no report card keep a stable alphabetical order behind the ranked ones.

The result leads with Trump, Li Qiang, Mishustin, Modi, Salman, Starmer, Merz — close to what Optimitron's expected-value task ranker produces, without the task tree that ranker needs. `compareSignersByMilitarySpending` is extracted and unit-tested, including that it reports a tie (so `Array.prototype.sort`'s stability preserves the query order) and that sorting a reversed roster lands on the same sequence.

## Before → After

| | Optimitron `/employees` | warondisease `/employees` |
| --- | --- | --- |
| Heading + mission statement | identical | identical |
| Treaty reminder composer | 4 steps, 3 recipient modes | identical |
| Program card (`RATIFY THE 1% TREATY`) with markdown, parameters, live counters | present | **dropped** |
| Signer roster | paginated table: assignee, task, deaths, dollars, time, remind | paginated list: assignee, overdue duration |
| Roster order | expected-value task ranking | military spending, then name |
| Rows per page | 10 | 10 |
| Signer count | 189 | 189 |
| Assignee link | `/tasks/1-pct-treaty-signer-xx` | `https://optimitron.com/tasks/...` (task pages stay in Optimitron per #238) |

## Ported into site-kit

The treaty reminder composer and the `ReminderComposer` shell it renders (extracted from Optimitron's 556-line `task-row-share`, which also holds the row dialog this PR does not need), plus the share-template, delay-attribution, task-accountability, share-channel, country-detection, hydrated-clock and request-origin modules. `retroui/*` swapped for the neobrutalist-ui equivalents.

The composer reads `session.user.countryCode` structurally: Optimitron's session carries it, the campaign apps' does not, and widening every app's NextAuth session type for one field this composer wants is the wrong trade. Absent means fall back to the browser locale.

## APIs

- `/api/government-leader` — direct port.
- `/api/share-attempts` — omits Optimitron's wish grant and user-treaty-task sync. Both write to gamification surfaces that exist only on optimitron.com and no campaign page reads them. Same divergence `/api/votes/sync` already has here.

Every `fetch()` in the ported closure was checked against `apps/warondisease/app/api`: those two plus `/api/referral-invitations`, which already existed.

## Snapshot

`/employees` is public but unlinked from the navigation — the same shape as `/soldiers` — so the nav walk never reaches it. It is declared in `additionalSnapshotRoutes`, and the overdue count and each row's duration are marked `data-volatile` so the snapshot records the roster and its order rather than a day count that rewrites itself on every regeneration.

It is also indexed in the campaign search, which the `campaign page index` test enforces for every public page.

## Verification

- `pnpm typecheck:apps` (all 7 apps), Optimitron web typecheck, `pnpm --filter @apps/warondisease lint` — clean
- `pnpm test:apps:unit` — 64 warondisease tests pass, including 6 new ones for the comparator
- `pnpm test:site-app-navigation` — 8/8
- Page rendered against the managed dataset: 189 signers, 19 pages, identical roster order across repeated requests
- Desktop (1440) and mobile (390) full-page screenshots captured and inspected locally; no overlap, no broken styling, and the one long name (Salman bin Abdulaziz Al Saud) wraps cleanly on mobile. Not committed.

## Open question for you

`/employees` has no navigation entry, matching how #276 landed `/joke`, `/poster`, `/love`, `/door-to-door`, `/fix-ai`, and `/foundations`. Whether "Remind Presidents" earns a slot in the campaign nav is an information-architecture call, not a porting one — tell me and it is a one-line change.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01TgYULvPW89bMAV3C6mXmtG

## Summary by CodeRabbit

* **New Features**
  * Added the “Remind Presidents” page for reviewing treaty signers, overdue tasks, and country-specific impact details.
  * Added customizable reminder messages with sharing via email, SMS, WhatsApp, Telegram, and social channels.
  * Added referral invitations, authenticated share tracking, government leader information, pagination, filtering, flags, and overdue impact metrics.

* **Bug Fixes**
  * Improved invitation validation, duplicate protection, message delivery limits, and country detection.

* **Tests**
  * Added coverage for referrals, sign-in requirements, country parsing, clock handling, filtering, and signer displays.

### CODE REVIEW

**Comment by mikepsinn** [comment]
Updated to 1c972522: merged `main` (82b9a2f4, which picked up #278) and registered `/employees` in the visual review.

**Heads up — `main` was red when I merged it in.** #278's "every site-app page is captured by the visual review or has a documented exemption" check and #276's eight new public pages merged 55 minutes apart, so neither PR's CI ever saw the other's contribution. `pnpm test:site-app-navigation` fails on `main` at 82b9a2f4 with eight missing pages.

I opened #287 for that — one file, registration only — and merged its branch into this one so this PR is green rather than red for a reason that has nothing to do with it. So the diff here carries #287's commit (a97a6cda) as well as its own.

**Merge order does not matter.** If #287 goes in first, this branch already contains that commit and the merge is trivial. If this one goes in first, it carries both changes and #287 becomes empty.

`/employees`'s own entry covers the page plus `treaty-reminder-composer.tsx`, `overdue-signer-list.tsx`, and `reminder-composer.tsx`, so a change to the composer or the roster re-flags this page.

Re-verified after both merges: `pnpm test:site-app-navigation` 16/16, `@apps/warondisease` typecheck clean, 64/64 unit tests.

File: packages/site-kit/src/lib/tasks/treaty-signers.server.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * Every treaty-signer task, most overdue first.
+ *
+ * `taskKey` is unique and every signer key shares one prefix, so this is a
+ * single index range scan. Tasks with no due date sort last — an unscheduled
+ * signature is not overdue, it is unscheduled — and the remaining ties break
+ * on assignee name so the roster is stable across renders.
+ */
+export async function getTreatySignerTasks(): Promise<TreatySignerTask[]> {
+  const rows = await prisma.task.findMany({
+    where: { taskKey: { startsWith: `${TREATY_SIGNER_TASK_KEY_PREFIX}:` } },
```
[the remark this answers is not in the document]
Comment (mikepsinn) [reply packages/site-kit/src/lib/tasks/treaty-signers.server.ts]: Fixed in ab700dd1a: signer tasks now require deletedAt: null and exclude VERIFIED status.

File: packages/site-kit/src/components/landing/treaty-reminder-composer.tsx [hunk 2]
```diff
@@ -0,0 +1,969 @@
...
+    ],
+  );
+
+  const recipientSelector = (
+    <div className="space-y-3">
+      <p className="text-xs font-black uppercase tracking-[0.14em] text-foreground">
+        Step 1. Select recipient
+      </p>
+      <div className="grid overflow-hidden border-2 border-foreground sm:grid-cols-3">
+        {(["president", "humanity", "one_human"] as const).map((mode, index) => {
```
[the remark this answers is not in the document]
Comment (mikepsinn) [reply packages/site-kit/src/components/landing/treaty-reminder-composer.tsx:758]: Fixed in ab700dd1a: signed-out One human selection now starts sign-in before exposing the authenticated workflow.

File: packages/site-kit/src/components/landing/treaty-reminder-composer.tsx [hunk 3]
```diff
@@ -0,0 +1,969 @@
...
+    const name = trimmedRecipientName;
+    if (!name) {
+      setError("Add a name before assigning the task.");
+      return null;
+    }
+
+    setIsCreatingInvitation(true);
+    setError(null);
+    try {
+      const created = await createReferralInvitationRequest({
```
[the remark this answers is not in the document]
Comment (mikepsinn) [reply packages/site-kit/src/components/landing/treaty-reminder-composer.tsx:456]: Fixed in ab700dd1a: the endpoint now accepts the shared recipientEmail contract, returns a singular invitation, and supports markCopied and sendMessage.

File: packages/site-kit/src/lib/detect-country.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+ * The region subtag (after the hyphen) maps directly to a country code.
+ * Returns null if no region subtag is present (e.g., "en", "fr").
+ */
+export function getCountryFromLocale(): string | null {
+  if (typeof navigator === "undefined") return null;
+  const lang = navigator.language;
+  if (!lang) return null;
+  const parts = lang.split("-");
+  const region = parts[1];
+  if (!region || region.length !== 2) return null;
```
[the remark this answers is not in the document]
Comment (mikepsinn) [reply packages/site-kit/src/lib/detect-country.ts]: Fixed in ab700dd1a: locale regions now come from Intl.Locale.region, with script-subtag regressions covered.

File: packages/site-kit/src/lib/tasks/accountability.ts [hunk 5]
```diff
@@ -0,0 +1,503 @@
...
+
+  const delayDalysLostPerDay = frame?.delayDalysLostPerDayBase ?? null;
+  const delayEconomicValueUsdLostPerDay = frame?.delayEconomicValueUsdLostPerDayBase ?? null;
+  const delayHumanLivesLostPerDay = derivePerDayMetric(frame, selectedMetrics, {
+    conditionalMetricKey: "lives_saved_if_success",
+    riskAdjustedMetricKey: "contribution_lives_saved_per_pct_point",
+  });
+  const delaySufferingHoursLostPerDay = derivePerDayMetric(frame, selectedMetrics, {
+    conditionalMetricKey: "suffering_hours_if_success",
+    riskAdjustedMetricKey: "contribution_suffering_hours_per_pct_point",
```
[the remark this answers is not in the document]
Comment (mikepsinn) [reply packages/site-kit/src/lib/tasks/accountability.ts:158]: Stylistic refactor outside this defect. These existing metric keys have an established contract and this comment identifies no triggering bug.

File: apps/warondisease/app/api/referral-invitations/route.ts [hunk 6]
```diff
@@ -268,13 +291,91 @@ export async function PATCH(req: NextRequest) {
...
+        )
+      }
+
+      const since = new Date(Date.now() - 24 * 60 * 60 * 1000)
+      const sendsToday = await prisma.referralInvitation.count({
+        where: { referrerUserId: userId, sentAt: { gte: since } },
+      })
+      if (sendsToday >= MAX_REFERRAL_INVITATION_EMAILS_PER_DAY) {
+        return NextResponse.json({ error: "Daily referral email limit reached" }, { status: 429 })
+      }
```
[the remark this answers is not in the document]
Comment (mikepsinn) [reply apps/warondisease/app/api/referral-invitations/route.ts:320]: Fixed in 0687682a6: already-sent invitations now return 409 before rate-limit counting, delivery, or email logging.

File: packages/site-kit/src/lib/tasks/impact.ts [hunk 7]
```diff
@@ -0,0 +1,410 @@
...
+  const selectedFrame =
+    availableFrames.find(
+      (frame) => normalizeFrameSlug(frame.frameSlug) === requestedSlug,
+    ) ??
+    availableFrames.find((frame) => frame.frameKey === requestedFrame) ??
+    availableFrames.find(
+      (frame) => frame.frameKey === DEFAULT_TASK_IMPACT_FRAME,
+    ) ??
+    availableFrames[0] ??
+    null;
```
[the remark this answers is not in the document]
Comment (mikepsinn) [reply packages/site-kit/src/lib/tasks/impact.ts:151]: Broad symmetry coverage request. Focused regressions now cover the actual president-board and frozen-clock failures without adding unrelated tests.

File: apps/warondisease/app/employees/page.logged-out.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+- Twitter title: The International Campaign to End War and Disease
+- Twitter description: Click a glowing rectangle. 15 seconds. 2.6 lives saved + 53 years of suffering prevented.
+
+## Visible Page Copy
+
+- [THE INTERNATIONAL CAMPAIGN TO END WAR AND DISEASE](/)
+- [Go to Dashboard](/dashboard)
+## PRESIDENT MANAGEMENT SYSTEM
+- You give these people $37 trillion a year. Their job is to promote the general welfare which means increasing median health and wealth. Your job is to remind them that this is their job. Please do your job by clicking the remind button.
+#### SEND EARTH OPTIMIZATION TASK REMINDER
```
[the remark this answers is not in the document]
Comment (mikepsinn) [reply apps/warondisease/app/employees/page.logged-out.md:20]: This file is generated copy-inventory evidence, not rendered Markdown. Changing the UI heading solely for markdownlint would alter the page's valid semantic hierarchy.

File: apps/warondisease/tests/unit/treaty-signers.test.ts [hunk 9]
```diff
@@ -0,0 +1,86 @@
+import { describe, expect, it } from "vitest"
+import {
+  compareSignersByMilitarySpending,
+  getAccountabilityReferenceMs,
+} from "@optimitron/site-kit/lib/tasks/treaty-signers"
+
+describe("getAccountabilityReferenceMs", () => {
```
[the remark this answers is not in the document]
Comment (mikepsinn) [reply apps/warondisease/tests/unit/treaty-signers.test.ts:7]: Not adding this test: `clientNowMs ?? serverNowMs` is a direct null-coalescing expression, and the rendered pre-hydration path already passes null. Existing tests cover the two meaningful behaviors—preventing clock rewind and accepting a later client time—so another assertion would only restate JavaScript semantics.

File: packages/site-kit/src/components/tasks/overdue-signer-list.tsx [hunk 10]
```diff
@@ -0,0 +1,495 @@
...
+                  overdueDays={overdueDays}
+                  task={task}
+                />
+              ))}
+            </ul>
+          ) : (
+            <p className="px-4 py-8 text-center text-sm font-black uppercase">
+              No president tasks match that filter.
+            </p>
+          )}
```
[the remark this answers is not in the document]
Comment (mikepsinn) [reply packages/site-kit/src/components/tasks/overdue-signer-list.tsx:463]: Fixed in 8612e185a: an empty unfiltered roster now says “No president tasks are open right now,” while an active search keeps the filter-specific message.
